### DISCUSSION TITLE
Various methods for barcode scanning in .NET MAUI

### DISCUSSION THREAD

**Discussion by giscus** [VladislavAntonyuk/vladislavantonyuk.github.io#116] [Announcements]
# Various methods for barcode scanning in .NET MAUI

This article explores three methods: using MLKit for Android, Camera.MAUI.ZXing, and CommunityToolkit.Maui.Camera.

https://vladislavantonyuk.github.io/articles/Various-methods-for-barcode-scanning-in-.NET-MAUI/

**Comment by cissemy** [comment]
Great job.
I am getting this error :
Xamarin.Google.MLKit.Common.MlKitException: 'Waiting for the Barcode UI module to be downloaded.'

**Reply by VladislavAntonyuk** [reply]
Hey,
You can find additional configuration for MLKit here: https://developers.google.com/ml-kit/vision/barcode-scanning/code-scanner.

So the answer for your comment: You can also explicitly check the scanner module availability and request download through Google Play services [ModuleInstallClient API](https://developers.google.com/android/guides/module-install-apis).

If you don't enable install-time module downloads or request explicit downloads—and the scanner module isn't already installed for another use case—Google Play services downloads the scanner module the first time it's used.

**Comment by hans-1995** [comment]
Very nice tutorial, thanks for sharing! ZXing is a decent open-source library, but struggles with low light and damaged barcodes. If you're looking for an enterprise-level barcode scanner, I'd recommend Scanbot SDK (disclaimer: I am part of the team). A colleague of mine recently wrote a similar tutorial to yours, which also includes the step-by-step integration of our SDK. I'll leave the link below. Let me know if you check it out. 

https://scanbot.io/techblog/zxing-maui-barcode-scanner-tutorial/

**Reply by VladislavAntonyuk** [reply]
ZXing also allows you to implement your barcode scanner and set it to ZxingCameraView.BarCodeDecoder
